### ISSUE TITLE
registered to event gets wrong firstname lastname of user but the right username

### ISSUE THREAD

**Issue by caperneoignis** [davidpesce/moodle-logstore_xapi#68]
Okay, not sure what is happening here. However, if I create a user, or if they register to a course one of two things happen. 

If you are using Totara HR import, it places the admin's name who is registering the user/creating them as the first name last name of the opts[] in the return array of events.php, so I'm assuming it is coming down from tincan expander with a different first name last name for opts. but the correct username. 

If the user self registers, it just blanks the first/last name like it is looking for something else or if it is the system registering them for automatic training, in the case of Totara. 

I haven't tried this on Moodle yet, especially the user upload feature. I already got a work around for my DataTables that uses the MongoDB connector to produce a json string for the DataTables to use to make a searchable table of statements. 

However, here is an example. 
{
    "version": "1.0.0",
    "actor": {
        "objectType": "Agent",
        "name": "Admin Doe",
        "account": {
            "name": "john.snuffy",
            "homePage": "https://example.com"
        }
    },

I have to redact the names, but the admin is me. My lastname is not Doe, just fyi ;). Not sure how to test this or what I should be looking for in the xapi plugin to rectify this, so the username and first.last name is the proper one. Let me know what you think, or if there is something I can try. This is getting inputted into Learning Locker 1.12.1 just in case that has a hand in this. 

I'll be more then happy to look into this, especially if it turns out this is a Totara only issue, and not based on the events coming from Moodle Core that Totara still has a lot internally. 

you can refer to issue: #60 on how I currently generate the first name last name for the actor.

**Comment by jlowe64** [comment]
Sorry you're having this issue. There are parts of totara's database that are fundamentally different from Moodle. This plugin is intended for Moodle and uses Moodle standard functions to get a lot of this information. Even so, totara database tables have many additional columns and changes that would make using this plugin on thier platform a bit of a challenge.

The expander is where we use some Moodle functions to get data. This would probably be where the issue is. As you already pointed out. 

I am unsure if totara has different functions to gather user data, but I would suspect they do.

**Comment by caperneoignis** [comment]
Alright, I'll look into it and let you know. I'll look at the way Totara handles registration events and see if there is any difference between that and Moodle. It just weird because the user name is fine, its the first and last name that is different. I'll see what I can find out.

**Comment by caperneoignis** [comment]
Okay, I think I figured out what is going on. in event.php I have 

``` javascript
'user_fullname' => $opts['user']->firstname.' '.$opts['user']->lastname,
```

However: 
I do not have that variable in EnrolmentCreated.php or UserRegistered.php so I added it to the two files. I'll test it here in a second with a test server, to see if the issue is resolved. 

``` javascript
'user_fullname' => $opts['relateduser']->firstname.' '.$opts['relateduser']->lastname,
```

This would explain why I'm getting a blank for user_fullname, because the variable does not exist, when a user registers for a course. Not sure in the case of user creation. I'll look for that event and see if it is missing there. I still keep user_name which is why I'm still getting the actual username under the account information. 

so It looks like its really my customization that has hosed this up, so I apologize for making the issue. 
EDIT: Files are located in moodle-xapi-translator\src\Events folder.
